relay/state.go

```diff
@@ -4,43 +4,82 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+
+	vaultType "github.com/vultisig/commondata/go/vultisig/vault/v1"
+	"github.com/vultisig/vultisigner/common"
 )
 
 type LocalStateAccessorImp struct {
 	Key    string
 	Folder string
+	Vault  *vaultType.Vault
 }
 
-func (l *LocalStateAccessorImp) ensureFolder() error {
+func (l *LocalStateAccessorImp) NewLocalStateAccessorImp(key, folder, vaultFileName, vaultPasswd string) error {
```

Comment:
Can you please make `NewLocalStateAccessorImp`  not part of `LocalStateAccessorImp`?  , just a function